app/logic/emailHandler.py

```diff
@@ -16,46 +16,52 @@
 from app.models.term import Term
 
 class EmailHandler:
-    def __init__(self, raw_form_data, url_domain, sender_object, attachment_file=None):
+    def __init__(self, raw_form_data, url_domain, attachment_file=[]):
 
         self.mail = Mail(app)
         self.raw_form_data = raw_form_data
         self.url_domain = url_domain
         self.override_all_mail = app.config['MAIL_OVERRIDE_ALL']
-        self.sender = sender_object
+        self.sender_username = None
+        self.sender_name = None
+        self.sender_address = None
+        self.reply_to = app.config['MAIL_REPLY_TO_ADDRESS']
         self.template_identifier = None
         self.subject = None
         self.body = None
-        self.reply_to = None
         self.event = None
-        self.program_id = None
+        self.program = None
         self.recipients = None
         self.sl_course_id = None
         self.attachment_path = app.config['files']['base_path'] + app.config['files']['email_attachment_path']
+        self.attachment_filepaths = []
         self.attachment_file = attachment_file
 
     def process_data(self):
         """ Processes raw data and stores it in class variables to be used by other methods """
         # Email Template Data
-        self.template_identifier = self.raw_form_data['templateIdentifier']
+        # Template Identifier
+        if 'templateIdentifier' in self.raw_form_data:
+            self.template_identifier = self.raw_form_data['templateIdentifier']
 
         if 'subject' in self.raw_form_data:
             self.subject = self.raw_form_data['subject']
 
-        if 'body' in self.raw_form_data:
-            self.body = self.raw_form_data['body']
-
-        if 'replyTo' in self.raw_form_data:
-            self.reply_to = self.raw_form_data['replyTo']
-
         # Event
         if 'eventID' in self.raw_form_data:
             event = Event.get_by_id(self.raw_form_data['eventID'])
             self.event = event
 
         # Program
-        if 'programID' in self.raw_form_data:
-            self.program_id = self.fetch_event_programs(self.raw_form_data['programID'])
+        if self.event:
+            self.program = Program.get_or_none(Program.id == self.event.program_id)
```

Comment:
You can just use `event.program` here, to get a program object or None.

Reply:
Great idea!